### PR TITLE
better Quality with direct_spatial_mv_pred_flag=1

### PR DESCRIPTION
**Pull request by QuPengfei**
this fix the R2R issue with AVC encoder

### CODE REVIEW

**Comment by scott-d-phillips** [comment]
I checked the intel/ci failures, they seem to be some setup issue unrelated to the the patches themselves. I believe it is OK to ignore these failures here.

**Comment by chivakker** [comment]
@MicheleL-Intel can you take a look at the failures?

**Comment by MicheleL-Intel** [comment]
It looks like the same issue with executing the containers for de9692ad47b.  Still investigating.

**Comment by MicheleL-Intel** [comment]
It's failing to build and execute the containers because media-ci2 crashed due to Java exception and proxy connection issues (see logs below).  It appears the build stopped abruptly.

Also observing warnings and errors in this particular commit as well, such as:
          20:38:45,700 WARN  - Base build not exist to calculate change sets. 

I'll need to inspect the server tomorrow as I'm not able to ssh to it.

=====================================
media-ci2 server log
=====================================
2017-03-29 22:25:59,564 [pool-1-thread-198075] ERROR com.pmease.quickbuild.DefaultBuildEngine - Error processing build request.
    java.lang.RuntimeException: Error executing check condition job.
        at com.pmease.quickbuild.CheckConditionTask.reduce(CheckConditionTask.java:39)
 	.....
        at java.lang.Thread.run(Thread.java:745)
    Caused by: com.pmease.quickbuild.QuickbuildException: Failed to run command: git remote prune origin 
    Command return code: 128
    Command error output: fatal: unable to access 'https://uartie:c703e5050d3f44d2ca6524580360018c33b6f373@github.com/01org/libyami/': Failed to connect to proxy-us.intel.com port 911: Connection timed out
        at com.pmease.quickbuild.execution.Commandline$ExecuteResult.buildException(Commandline.java:496)

**Comment by MicheleL-Intel** [comment]
Examined build agent logs and found an issue with the commit not being able to retrieve HEAD description.  Requested @QuPengfei to resubmit and looks like the build passes this time.  Another prior commit built successfully as well.

Will monitor for possible intermittent parallel build issue.

@xhaihao, regarding whether this is needed to fix the R2R issue, according to QuPengfei, this is supposed to fix it for AVC.

**Comment by xhaihao** [comment]
the root cause for the R2R issue is still unknown, please don't merge this commit

**Comment by QuPengfei** [comment]
@xhaihao it is good to get better quality with direct_spatial_mv_pred_flag=1. we trace the R2R issue on https://github.com/01org/intel-vaapi-driver/issues/84

File: encode/avcenc.c [hunk 1]
```diff
@@ -2051,7 +2056,10 @@ int main(int argc, char *argv[])
     alloc_encode_resource(yuv_fp);
 
     enc_frame_number = 0;
-    for ( f = 0; f < frame_number; f++) {		//picture level loop
+    if(frame_num_value == 0)
```
Comment (xhaihao) [inline encode/avcenc.c]: how about change the condition to 'frame_num_value <= 0'?

**Comment by xhaihao** [comment]
@QuPengfei  I'm fine to merge your 2nd patch if it can improve the quality

**Comment by QuPengfei** [comment]
Any comments about this patch?

**Comment by QuPengfei** [comment]
the branch is rebased and updated

**Pull request merged**